### PR TITLE
Streamline Settings

### PR DESCRIPTION
**Pull request by szaimen**
This PR is best reviewed like this: https://github.com/nextcloud/contacts/pull/2372/files?diff=unified&w=1

| Before | After |
|---|---|
| image | image |

Signed-off-by: szaimen <szaimen@e.mail.de>

### CODE REVIEW

**Comment by szaimen** [comment]
Every time when I try to compile the js it fails with 
`npm WARN tarball tarball data for cdav-library@git+ssh://git@github.com/nextcloud/cdav-library.git#e62c7c9942d6759743dc9a0350921e898440f2d5 (sha512-xgZ7MVeYeiXwQl5SuyXRVxXPM2swxG46HS0m0IFQ2SmzOgBTm3AdzUFJicggJ34e3SIzrFniWaftkhr550eyJw==) seems to be corrupted.`

**Comment by skjnldsv** [comment]
Requires @jancborchardt approval.
But there was different plans for the settings: https://github.com/nextcloud/contacts/issues/1703#issuecomment-703083443

**Comment by szaimen** [comment]
> But there was different plans for the settings: [#1703 (comment)](https://github.com/nextcloud/contacts/issues/1703#issuecomment-703083443)

Yes but until it isn't inside a modal which I am not able to do myself, I thought this was a good small fix which improves the readability of the settings by a lot, imho.

File: src/components/AppNavigation/SettingsSection.vue
Comment (artonge) [inline src/components/AppNavigation/SettingsSection.vue:26]: Should use [BEM naming style](https://cssguidelin.es/#bem-like-naming)

```suggestion
 		<div v-if="allowSocialSync" class="social-sync__list-entry">
```

File: css/SettingsSection.scss
Comment (artonge) [inline css/SettingsSection.scss:43]: Not a fan of specifying width en elements that do not need to be constrained. Can't the browser guess that width from the parent ?

File: css/SettingsSection.scss
Comment (artonge) [inline css/SettingsSection.scss:37]: Could be grouped with other icon styles in a shared class like `settings-line__icon` or something.

**Review by artonge — CHANGES_REQUESTED** [review]

File: src/components/AppNavigation/SettingsSection.vue
Comment (szaimen) [reply src/components/AppNavigation/SettingsSection.vue]: done

File: css/SettingsSection.scss
Comment (szaimen) [reply css/SettingsSection.scss]: done

**Comment by jancborchardt** [comment]
Hmm, so since these icons would not be there when moving the settings to the modal, I would say we don’t need to add them here. (And also e.g. the icon is better inside the "Import contacts" button than outside it. :)

**Comment by szaimen** [comment]
> Hmm, so since these icons would not be there when moving the settings to the modal, I would say we don’t need to add them here.

The question is how long it will take to bring the settings into the Modal since the current settings are really bad from a        usability perspective, imo. So I'd rather merge this and revert it when bringing it into the settings modall than living with it for another release 😅

File: css/SettingsSection.scss
Comment (szaimen) [reply css/SettingsSection.scss:39]: that doesn't work unfortunately.

**Comment by szaimen** [comment]
> (And also e.g. the icon is better inside the "Import contacts" button than outside it. :)

I've reverted that change, see updated screenshot.

**Comment by nimisha-vijay** [comment]
If we are moving to a modal soon then these changes will not be in effect for too long, but I do think the icons look good. Couple of suggestions:
- move the content of the settings menu slightly to the left so that the checkbox and icons in the content are left aligned with the settings icon in the header, and all the text of the header and content is left aligned as well
- "Enter new address book name" can be changed to something shorter like "Add new address book"?

**Comment by szaimen** [comment]
@nimisha-vijay thanks for the feedback! :)
Addressed both points, see updated screenshots

**Review by artonge — APPROVED** [review]
OK, code wise

**Comment by jancborchardt** [comment]
Ok – then I just have 2 more pieces of feedback :)
- The sorting setting would be better as the first item in the settings, as it’s the most basic one.
- The address books could do with a header "Address books" to actually understand what they are. (This heading should also have some whitespace above so that section is a bit separated from the two simpler settings above.)

**Comment by szaimen** [comment]
Done. Please have another look at the updated screenshots @jancborchardt and @nimisha-vijay :)

**Review by jancborchardt — APPROVED** [review]

**Comment by szaimen** [comment]
@artonge do you mind reviewing the code another time? I made some small changes...

**Review by Pytal — APPROVED** [review]

**Pull request merged**
